packages/cozy-mespapiers-lib/src/utils/search.js

```diff
@@ -0,0 +1,120 @@
+import { Document } from 'flexsearch'
+
+import { models } from 'cozy-client'
+
+import { CONTACTS_DOCTYPE } from '../doctypes'
+
+const {
+  document: {
+    themes: { themesList }
+  },
+  file: { isFile }
+} = models
+
+export const index = new Document({
+  document: {
+    id: '_id',
+    tag: 'flexsearchProps:tag',
+    index: [
+      'flexsearchProps:name',
+      'flexsearchProps:translatedQualificationLabel',
+      'flexsearchProps:fullname',
+      'flexsearchProps:birthday',
+      'flexsearchProps:birthcity',
+      'flexsearchProps:email',
+      'flexsearchProps:civility',
+      'flexsearchProps:company',
+      'flexsearchProps:jobTitle'
+    ],
+    store: ['_id', '_type']
+  }
+})
+
+export const makeFileTags = file => {
+  const item = file.metadata.qualification
+  const tags = themesList
+    .filter(theme => {
+      return theme.items.some(it => it.label === item.label)
+    })
+    .map(x => x.label)
+  return tags
+}
+
+export const makeContactTags = contact => {
+  const contactTags = []
+
+  const themesByAttributes = {
+    givenName: ['identity'],
+    familyName: ['identity'],
+    phone: ['home', 'work_study', 'identity'],
+    email: ['work_study', 'identity'],
+    cozy: ['identity'],
+    address: ['home', 'work_study', 'identity'],
+    birthday: ['identity'],
+    company: ['work_study'],
+    jobTitle: ['work_study']
+  }
+
+  Object.keys(themesByAttributes).map(attribute => {
+    const preAttribute = ['givenName', 'familyName'].includes(attribute)
+      ? 'name'
+      : undefined
+
+    const value = preAttribute
+      ? contact?.[preAttribute]?.[attribute]
+      : contact?.[attribute]
+
+    const themes = themesByAttributes[attribute]
+
+    if (value && value.length > 0) {
+      themes.forEach(theme => {
+        if (!contactTags.includes(theme)) {
+          contactTags.push(theme)
+        }
+      })
+    }
+  })
+
+  return contactTags
+}
+
+export const addFileDoc = (index, doc, scannerT) => {
+  return index.add({
+    ...doc,
+    flexsearchProps: {
+      tag: makeFileTags(doc),
+      name: doc.name,
+      translatedQualificationLabel: scannerT(
+        `items.${doc.metadata.qualification.label}`
+      )
```

Comment:
It looks like you duplicate all the indexed fields in the "virtual" `flexsearchProps` object - why is it needed? I fear this duplication will increase the index size and hurt the performances

Reply:
cf https://github.com/cozy/cozy-libs/pull/2029#discussion_r1126411704 non ?